plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -971,21 +976,20 @@ def try_get_gas_pricing(
         finally:
             self._api.eth.set_gas_price_strategy(prior_strategy)  # pragma: nocover
 
-        if old_tip is not None and gas_price_strategy is EIP1559:
```

Comment:
The first issue with the repricing was when checking which strategy to use since the strategy was `None`. As a result, the calculated price was always returned. This was fixed in e2bcbeb, by getting it with the callable from `_get_gas_price_strategy_callable` (now renamed as `_get_gas_price_strategy`).